src/comparator/file_set_comparator.py

```diff
@@ -46,27 +46,85 @@ def compare(self):
+            return
+        api_version_original = fs_original.api_version
+        api_version_update = fs_update.api_version
         for option in packaging_options_original.keys():
-            language_option_original = set(packaging_options_original[option].keys())
-            language_option_update = set(packaging_options_update[option].keys())
-            for language_option in language_option_original - language_option_update:
-                removed_option = packaging_options_original[option][language_option]
-                self.finding_container.addFinding(
-                    category=FindingCategory.PACKAGING_OPTION_REMOVAL,
-                    proto_file_name=removed_option.proto_file_name,
-                    source_code_line=removed_option.source_code_line,
-                    message=f"An exisiting packaging option for `{option}` is removed.",
-                    actionable=True,
-                )
-            for language_option in language_option_update - language_option_original:
-                added_option = packaging_options_update[option][language_option]
-                self.finding_container.addFinding(
-                    category=FindingCategory.PACKAGING_OPTION_ADDITION,
-                    proto_file_name=added_option.proto_file_name,
-                    source_code_line=added_option.source_code_line,
-                    message=f"An exisiting packaging option for `{option}` is added.",
-                    actionable=True,
+            per_language_options_original = list(
+                packaging_options_original[option].keys()
+            )
+            per_language_options_update = list(packaging_options_update[option].keys())
+            # Compare the option of `java_outer_classname`. No minor version updates
+            # need to consider.
+            if option == "java_outer_classname":
+                # Sort the java_outer_classname list for comparison.
+                classnames_original = sorted(
+                    per_language_options_original, key=str.lower
                 )
+                classnames_update = sorted(per_language_options_update, key=str.lower)
+                for i, classname in enumerate(classnames_original):
+                    if i >= len(classnames_update):
+                        classname_option = packaging_options_original[option][classname]
+                        self.finding_container.addFinding(
+                            category=FindingCategory.PACKAGING_OPTION_REMOVAL,
+                            proto_file_name=classname_option.proto_file_name,
+                            source_code_line=classname_option.source_code_line,
+                            message=f"An exisiting packaging option `{classname}` for `{option}` is removed.",
+                            actionable=True,
+                        )
+                    elif classname != classnames_update[i]:
```

Comment:
Is it possible that `classnames_update` has more items than `classnames_original`, and these additional items are put in the front after sorting? For instance, `classnames_original` has `ServiceProto2`, but `classnames_update` has `ServiceProto1, ServiceProto2`. Since the code compares the items one by one, is this considered a `PACKAGING_OPTION_CHANGE` or breaking change?

Looking at the test case `file_options_original.java_outer_classname = "ServiceProto"`, it seems the `classnames` will have at most one element?

Reply:
1. yes, I think this case is valid, thanks for catching this, make the classnames a set should resolve this problem. Will fix it.
2. No, for a single file, `java_outer_classname` is one element, while for a file_set which contains a set of proto files information, the `java_outer_classnames` will be multiple. So i use a map to track them, key is the classname, value is the WithLocation object with the classname and the source code info.

Reply:
Seems this should be fixed in #82 and not related to this PR, so I will make a separate fix as next PR. Thanks!

Reply:
sgtm, thanks!